lib/src/kbasesearchengine/main/SearchMethods.java

```diff
@@ -282,10 +283,13 @@ public SearchObjectsOutput searchObjects(SearchObjectsInput params, String user)
     @Override
     public GetObjectsOutput getObjects(final GetObjectsInput params, final String user)
             throws Exception {
-
+        final Set<Integer> accessGroupIDs;
         final long t1 = System.currentTimeMillis();
-        final Set<Integer> accessGroupIDs =
-                new HashSet<>(accessGroupProvider.findAccessGroupIds(user));
+        if (user != null) {
```

Comment:
User cannot be null, dead code

Should probably have a null check 

Note there's still a bug here in that authed users can't get access to public data unless it's in a workspace to which they have explicit read perms, but that's outside the scope of this PR